src/FeaturesViewer.cpp

```diff
@@ -291,209 +260,111 @@ namespace qtAliceVision
+      // (5) Endpoints
+      geometryEndpoint = appendChildGeometry(node, nbEndpointsToDraw);
     }
     else
     {
-      if (oldNode->childCount() < 5)
-        return;
-
-      auto* rootHighlightPoint = static_cast<QSGGeometryNode*>(oldNode->childAtIndex(1));
-      auto* rootTrackLine = static_cast<QSGGeometryNode*>(oldNode->childAtIndex(2));
-      auto* rootReprojectionErrorLine = static_cast<QSGGeometryNode*>(oldNode->childAtIndex(3));
-      auto* rootPoint = static_cast<QSGGeometryNode*>(oldNode->childAtIndex(4));
-
-      if (!rootHighlightPoint || !rootTrackLine || !rootReprojectionErrorLine || !rootPoint)
-        return;
-      
-      rootHighlightPoint->markDirty(QSGNode::DirtyGeometry);
-      rootTrackLine->markDirty(QSGNode::DirtyGeometry);
-      rootReprojectionErrorLine->markDirty(QSGNode::DirtyGeometry);
-      rootPoint->markDirty(QSGNode::DirtyGeometry);
-
       // (1) Highlight points
-      geometryHighlightPoint = rootHighlightPoint->geometry();
-      geometryHighlightPoint->allocate(
-        static_cast<int>(nbHighlightPointsToDraw),
-        static_cast<int>(0)
-      );
+      geometryHighlightPoint = getCleanChildGeometry(node, 1, nbHighlightPointsToDraw);
       // (2) Tracks lines
-      geometryTrackLine = rootTrackLine->geometry();
-      geometryTrackLine->allocate(
-        static_cast<int>(nbTrackLinesToDraw * kLineVertices),
-        static_cast<int>(0)
-      );
+      for (std::size_t idx = 0; idx < 3; idx++) 
+        geometryTrackLine[idx] = getCleanChildGeometry(node, idx+2, nbTrackLinesToDraw[idx] * kLineVertices);
       // (3) Reprojection Error lines
-      geometryReprojectionErrorLine = rootReprojectionErrorLine->geometry();
-      geometryReprojectionErrorLine->allocate(
-        static_cast<int>(nbReprojectionErrorLinesToDraw * kLineVertices),
-        static_cast<int>(0)
-      );
+      geometryReprojectionErrorLine = getCleanChildGeometry(node, 5, nbReprojectionErrorLinesToDraw * kLineVertices);
       // (4) Points
-      geometryPoint = rootPoint->geometry();
-      geometryPoint->allocate(
-        static_cast<int>(nbPointsToDraw),
-        static_cast<int>(0)
-      );
+      geometryPoint = getCleanChildGeometry(node, 6, nbPointsToDraw);
+      // (5) Endpoints
+      geometryEndpoint = getCleanChildGeometry(node, 7, nbEndpointsToDraw * kTriangleVertices);
     }
 
     geometryHighlightPoint->setDrawingMode(QSGGeometry::DrawPoints);
     geometryHighlightPoint->setLineWidth(6.0f);
 
-    geometryTrackLine->setDrawingMode(QSGGeometry::DrawLines);
-    geometryTrackLine->setLineWidth(2.0f);
+    geometryTrackLine[0]->setDrawingMode(QSGGeometry::DrawLines);
+    geometryTrackLine[0]->setLineWidth(2.0f);
+
+    geometryTrackLine[1]->setDrawingMode(QSGGeometry::DrawLines);
+    geometryTrackLine[1]->setLineWidth(2.0f);
+
+    geometryTrackLine[2]->setDrawingMode(QSGGeometry::DrawLines);
+    geometryTrackLine[2]->setLineWidth(5.0f);
```

Comment:
Note: the last call to the function `setLineWidth` seems to affect all the line width of the 3D Viewer as well.
Something to fix here or in Meshroom 3D Viewer.